app/controllers/games_controller.rb

```diff
@@ -1,17 +1,19 @@
 class GamesController < ApplicationController
+  include GamesHelper
+
   def index
+    @user = User.create
     @games = Game.all
   end
 
   def create
     @game = Game.create
-    @game.join(@user)
+    @game.users.add(current_user)
 
     render action: show
   end
 
   def show
     @game = Game.find(params[:id])
-    @game.join(@user)
   end
```

Comment:
To join a game we should use the game#update action, where we can set the user id. This can easily be done the default rails way with a form helper in the view